cli/command/container/cp.go

```diff
@@ -98,14 +94,18 @@ func NewCopyCommand(dockerCli command.Cli) *cobra.Command {
 			}
 			opts.source = args[0]
 			opts.destination = args[1]
+			if !cmd.Flag("quiet").Changed {
+				// User did not specify "quiet" flag; suppress output if no terminal is attached
+				opts.quiet = !dockerCli.Out().IsTerminal()
+			}
```

Comment:
This is a bit "quirky" but fixes the `docker cp --help | grep quiet` case (which is confusing), but still allows automatic detection _if_ the user didn't explicitly set `--quiet` / `--quiet=true` / `--quiet=false`

Reply:
Good point, works beauty 😄